src/Data/Gdpr/GdprDataClass.ts

```diff
@@ -9,6 +10,77 @@ export const Gdpr = provideDataClass(
       return (await import('./gdprParamsFallback')).gdprParamsFallback()
     }
 
-    return { restApi }
+    // TODO: simply return { restApi }, once #11616 is resolved
+    if (!restApi.headers.Authorization) return { restApi }
+
+    return {
+      connection: {
+        async index(params) {
+          if (Object.keys(params.filters()).length > 0) {
+            throw new Error('GDPR consent does not support filtering.')
+          }
+
+          if (params.order().length > 0) {
+            throw new Error('GDPR consent does not support sorting.')
+          }
+
+          if (params.search()) {
+            throw new Error('GDPR consent does not support searching.')
+          }
+
+          const urlParams = new URLSearchParams()
+
+          const limit = params.limit()
+          if (limit) urlParams.set('_limit', limit.toString())
+
+          const continuation = params.continuation()
+          if (continuation) urlParams.set('_continuation', continuation)
+
+          const url = `${restApi.url}?${urlParams.toString()}`
+
+          const response = await fetch(url, { headers: restApi.headers })
+          if (!response.ok) {
+            const error = new DataConnectionError(
+              'Failed to fetch GDPR consent',
+            )
+            errorToast('Unable to connect to PisaSales', error)
+            throw error
+          }
+
+          return response.json()
+        },
+        async update(id, data) {
+          const response = await fetch(`${restApi.url}/${id}`, {
+            method: 'PATCH',
+            headers: { ...restApi.headers, 'Content-Type': 'application/json' },
+            body: JSON.stringify(data),
+          })
+          if (!response.ok) {
+            const error = new DataConnectionError(
+              'Failed to update GDPR consent',
+            )
+            errorToast('Unable to connect to PisaSales', error)
```

Comment:
* [ ] "Unable to connect to PisaSales" is nothing a visitor would understand. "Failed to update GDPR consent" is fine though.

Reply:
I think a good compromise would be to have both:
* An error message that helps the user understand they could contact support. "Unable to connect to PisaSales" is not good at explaining the next step. A visitor would probably go and search the web for "PisaSales" or "How can I connect to PisaSales", which would lead them nowhere.
* Error details a supporter can work with. Mentioning PisaSales is not necessarily helpful here, b/c the responsible backend is probably already known at the receiving end.